app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt

```diff
@@ -16,8 +16,9 @@ class Example1Activity : BaseMvpActivity<Example1Contract.Presenter>(), Example1
         return Example1Presenter(this, configuration)
     }
 
+    @Suppress("ProtectedInFinal")
     @OnClick(R.id.example1_button)
-    internal fun onButtonClick() {
+    protected fun onButtonClick() {
```

Comment:
I personally really like this pattern for consuming generated java code from kotlin, but I haven't seen it espoused elsewhere so I want to flag it for discussion here (it's also come up in slack a couple of times).

This method needs to be visible to Butterknife, but otherwise we don't want it exposed. The idiomatic pure-Kotlin way to do so would be to make it `internal`, and that's what the linter will push you toward. `protected` members in Kotlin are visible in Kotlin only to subclasses, and not to other code in the same package. However, Java code in the same package can access these things, and that's what Butterknife generates. Therefore making this `protected` hides this method from all of our other Kotlin code while still letting Butterknife do it's thing. The Kotlin linter thinks there's no reason to have a protected member in a final class because it doesn't consider this interop case, so we suppress the warning it generates.

The same pattern works for Dagger `@Inject` items, among other things.

Reply:
yeah, protected sounds better to me too. it's a shame kotlin doesn't support package protected, this is one cons for kotlin compared to java

Reply:
Lets remove butterknife and use synthetic accessors

Reply:
Unless we go the full Anko route, I still like butterknife for listeners, even if you want to use kotlinx.android for view references.